packages/puppeteer-core/src/common/Binding.ts

```diff
@@ -0,0 +1,107 @@
+import {isErrorLike} from '../util/ErrorLike.js';
+import {ExecutionContext} from './ExecutionContext.js';
+
+/**
+ * @internal
+ */
+export class Binding {
+  #name: string;
+  #fn: (...args: unknown[]) => unknown;
+  constructor(name: string, fn: (...args: unknown[]) => unknown) {
+    this.#name = name;
+    this.#fn = fn;
+  }
+
+  /**
+   * @param throwError - Whether to throw the error instead of sending it to the
+   * binding.
+   */
+  async call(
```

Comment:
nit: perhaps the name of this method could be `onCall` or smth like this to indicate that the binding is not invoked here but it is the logic that happens after the page called the binding? Although, I could see how `call` could reflect that but I think it is also more confusing.